lib/Github/HttpClient/CachedHttpClient.php

```diff
@@ -105,4 +114,31 @@ public function getLastResponse($force = false)
 
         return ($force) ? $lastResponse : $this->lastCachedResponse;
     }
+
+    /**
+     * Format the path and add query parameters if they exist.
+     *
+     * @param string $path
+     * @param array  $options
+     * @return void
+     */
+    private function formatPath($path, array $options)
+    {
+        $this->path = $path;
+
+        if (array_key_exists('query', $options) && !empty($options['query'])) {
+            $this->path .= '?';
+
+            $i = 0;
+            foreach ($options['query'] as $key => $value) {
```

Comment:
I know it will break :)

https://github.com/KnpLabs/php-github-api/blob/master/doc/request_any_route.md
The doc never says one has to encode by itself each query parameters.

It is very important because if we urlencode following your suggestion then if someone doesn't encode by hand the query parameters with the custom query builder then the cache won't work properly again, the md5 will be different. It is really a pain in the ass!

So:
- Url encode when $option['query'] is not empty?
(you proposed a code sample **BUT** we can just use `$request->getQuery()` because Guzzle has already encoded the query parameters...)
- Update the documentation to tell users to encode by hand?

Question:
You said it looks like a bug if the query parameters are not encoded. Why it is working so?